app/views/hosts/show.html.erb

```diff
@@ -1,98 +1,91 @@
-      cpu_allocation = @host[:summary].hardware.numCpuCores * @host[:summary].hardware.cpuMhz
-      memory_allocation = @host[:summary].hardware.memorySize / (1024 ** 2)
-      memory_usage = @host[:summary].quickStats.overallMemoryUsage
-      memory_usage_percentage = 0
-      cpu_usage_percentage = 0
-
-
-      if cpu_allocation != 0
-        cpu_usage_percentage = (cpu_usage / cpu_allocation.to_f * 100).round()
-      end
-
-      if memory_allocation != 0
-        memory_usage_percentage = (memory_usage / memory_allocation.to_f * 100).round()
-      end
-    end
-   %>
-
-    <tr>
-    <td>CPU usage:</td>
-    <td><%= cpu_usage_percentage %> %</td>
-    </tr>
-
-    <tr>
-    <td>Memory Usage:</td>
-    <td><%= memory_usage_percentage %> % (<%= memory_usage %> / <%= memory_allocation %> MB)</td>
-    </tr>
-
-    <tr>
-    <td>VMs:</td>
-    <td>
-    <% @host[:vm_names].each do |vm| %>
-        <%= link_to vm, controller: "vms", action: "show", id: vm %>
-        <br />
-    <% end unless @host.nil?%>
+  <tr>
+      <% up_time = seconds_to_units(Time.current - @host[:bootTime]) unless @host[:bootTime].nil? %>
+      <% state =  @host[:connectionState] =='connected' %>
+  
+      <td class="lead align-middle" colspan="2">
+        <div class="<%= state ? 'bg-success' : 'bg-danger' %> text-white float-left pb-1 px-4 mr-2">
+          <%= state ? 'online' : 'offline' %>
+        </div>
+        <%= " for #{up_time}" if state %>
     </td>
-    </tr>
-    <tr>
-      <td><%= button_to "Reserve Timeslot",
-                        "",
-                        method: :get, class: "btn btn-info" %></td>
-    </tr>
-
-  </tbody>
+   <td class="text-right" style="width: 20%">
+     <div class="dropdown border text-right">
+        <a class="dropdown-toggle text-muted ml-0 pr-2 pt-1 lead nav-link" href="#" id="navbarDropdown" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" aria-disabled="true">
+          Manage
+        </a>
+        <div class="dropdown-menu" aria-labelledby="navbarDropdown">
+            <a class="dropdown-item" href="#">Reserve Timeslot</a>
+        </div>
+      </div>
+    </td>
+  </tr>
 </table>
+<% 
+
+  hdd_free_space = (@host[:summary]&.host&.datastore.sum{ |datastore|  datastore.summary.freeSpace }.to_f / 1024**3).round(2) || 0
+  hdd_allocation = (@host[:summary]&.host&.datastore.sum{ |datastore|  datastore.summary.capacity }.to_f / 1024**3).round(2) || 0
+  hdd_usage = (hdd_allocation - hdd_free_space).round(2)
+  cpu_usage = @host[:summary].quickStats&.overallCpuUsage || 0
+  cpu_cores = @host[:summary]&.hardware&.numCpuCores || 1
+  cpu_mhz = @host[:summary]&.hardware&.cpuMhz || 0
+  cpu_allocation = cpu_cores * cpu_mhz
+  ram_usage = (@host[:summary].quickStats&.overallMemoryUsage.to_f / 1024).round(2) || 0
+  ram_allocation = (@host[:summary]&.hardware&.memorySize.to_f / 1024**3).round(2) || 0
+  cpu_cores = @host[:summary]&.hardware&.numCpuCores  || 1
```

Comment:
Is there a reason why `cpu_cores` is set a second time? :slightly_smiling_face:

Reply:
Yes. The API does not need to provide the value according to its doc. Therefore, we catch the case that any attribute (also in between) is not set.

Reply:
I mean literally two times, as in line `46` and `41` :grin: Having some sort of default value in case the API does not provide anything also seems sensible to me.

Reply:
Thank you, I deleted the second assignment :D

Reply:
:+1: